### PR TITLE
Add CalMAPPER existing projects data.

### PR DESCRIPTION
**Pull request by elsiel23**
Screen Shot 2022-10-31 at 2 39 30 PM

### CODE REVIEW

File: src/interface/src/app/map/map.component.ts [hunk 1]
```diff
@@ -41,28 +41,38 @@ export class MapComponent implements AfterViewInit {
...
-  //       },
-  //       onEachFeature: function(feature, layer) {
-  //         layer.bindPopup('Name: ' + feature.properties.PROJECT_NAME + '<br>' +
-  //         'Status: ' + feature.properties.PROJECT_STATUS);
-  //       }
-  //     }
-  //   ).addTo(this.map);
+    const searchParams = new URLSearchParams(params).toString();
+    var url = "https://services1.arcgis.com/jUJYIo9tSA7EHvfZ/ArcGIS/rest/services/CMDash_v3_view/FeatureServer/2/query?" + searchParams;
+    console.log(url)
```
Comment (thequeenofspades) [inline src/interface/src/app/map/map.component.ts:53]: Remove?

File: src/interface/src/app/map/map.component.ts [hunk 2]
```diff
@@ -41,28 +41,38 @@ export class MapComponent implements AfterViewInit {
...
-  //         }
-  //       },
-  //       onEachFeature: function(feature, layer) {
-  //         layer.bindPopup('Name: ' + feature.properties.PROJECT_NAME + '<br>' +
-  //         'Status: ' + feature.properties.PROJECT_STATUS);
-  //       }
-  //     }
-  //   ).addTo(this.map);
+    const searchParams = new URLSearchParams(params).toString();
+    var url = "https://services1.arcgis.com/jUJYIo9tSA7EHvfZ/ArcGIS/rest/services/CMDash_v3_view/FeatureServer/2/query?" + searchParams;
```
Comment (thequeenofspades) [inline src/interface/src/app/map/map.component.ts:52]: Pull into a constant?

File: src/interface/src/app/map/map.component.ts [hunk 3]
```diff
@@ -41,28 +41,38 @@ export class MapComponent implements AfterViewInit {
 
   constructor(private boundaryService: BoundaryService, private popupService: PopupService) {}
 
-  // // TODO(elsieling) Port data from django backend to this component
-  // private initCalMapperLayer() {
-  //   var data = JSON.parse(existing_projects)
+  private async initCalMapperLayer(layers: L.Control.Layers) {
```
Comment (thequeenofspades) [inline src/interface/src/app/map/map.component.ts:44]: A comment explaining (in brief) what this function does and what the CalMapper layer is would be helpful, I think.

**Review by thequeenofspades — COMMENTED** [review]
I pulled the changes from this branch and tested it out. Looks good, I can click on the projects and trigger the popups, and the performance doesn't seem too bad. It's also nice that Leaflet has a built-in control for layers, even though we'll probably be replacing it with our own component.

My general feeling is that most of the logic currently in map.component.ts actually belongs in an Angular service. Generally business logic should be kept out of components and confined to services. I also think some of this logic should probably be done on the backend and fetched via http call--I think Jon was working on that, so maybe let's loop him in.

But I'm also OK with these changes getting merged for now, and then Cynthia or I can go in and refactor the logic into a service, since that's a front-end task. And Jon can move whatever logic makes sense to move to the backend. WDYT?

File: src/interface/src/app/map/map.component.ts (hunk 1 above)
Comment (thequeenofspades) [reply src/interface/src/app/map/map.component.ts:53]: (just line 53)

**Comment by elsiel23** [comment]
That makes sense to me. I'm not very familiar with Angular, so I'll defer to you and Cynthia on how you want to set up the services and then I can move things over. 

I'll ask Jon about the HTTP interface!

**Review by thequeenofspades — APPROVED** [review]

**Comment by elsiel23** [comment]
Discussed a bit with Jon. I'll merge this for now and then work on pulling the query into a Django app that the Angular component can call.

**Pull request merged**